src/App.tsx

```diff
@@ -1,61 +1,138 @@
+import { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { User } from './types/User';
+
+function findUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(userItem => userItem.id === userId);
+
+  return foundUser || null;
+}
+
+const initialTodos = todosFromServer.map(todo => ({
+  ...todo,
+  user: findUser(todo.userId),
+}));
 
 export const App = () => {
+  const [titleError, setTitleError] = useState(false);
+  const [userIdError, setUserIdError] = useState(false);
+  const [todos, setTodos] = useState(initialTodos);
+  const [todoInfo, setTodoInfo] = useState({
+    id: 0,
+    title: '',
+    userId: 0,
+    user: null,
+  });
+
+  const { title } = todoInfo;
+
+  const listUser = usersFromServer;
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!todoInfo.title.trim() || todoInfo.userId === 0) {
+      if (!todoInfo.title.trim()) {
+        setTitleError(true);
+      }
+
+      if (todoInfo.userId === 0) {
```

Comment:
```suggestion
      if (!todoInfo.userId) {
```
**PLS DO THE SAME FOR ALL SUCH CASES IN THE PROJECT**